### PR TITLE
Implement [M3C] Wonderscape Sage

### PR DESCRIPTION
**Pull request by grimreap124**
https://scryfall.com/card/m3c/49/wonderscape-sage

### CODE REVIEW

**Comment by grimreap124** [comment]
I need to update to check if the land had a nonbasic land type as it existed on the battlefield


Wonderscape Sage's last ability doesn't care if the returned land was nonbasic. It only cares if the land had a nonbasic land type. For example, if you control Omo, Queen of Vesuva and return a Forest with an everything counter on it to your hand to pay the cost of Wonderscape Sage's last ability, you won't discard a card.

Use the subtypes of the returned land as it last existed on the battlefield to determine whether or not you must discard a card. For example, if you return a Copy Land that's a copy of Urza's Cave (a land with the types Urza's and Cave) to your hand to pay the cost of Wonderscape Sage's last ability, you won't discard a card.

**Comment by grimreap124** [comment]
> I need to update to check if the land had a nonbasic land type as it existed on the battlefield
> 
> Wonderscape Sage's last ability doesn't care if the returned land was nonbasic. It only cares if the land had a nonbasic land type. For example, if you control Omo, Queen of Vesuva and return a Forest with an everything counter on it to your hand to pay the cost of Wonderscape Sage's last ability, you won't discard a card.
> 
> Use the subtypes of the returned land as it last existed on the battlefield to determine whether or not you must discard a card. For example, if you return a Copy Land that's a copy of Urza's Cave (a land with the types Urza's and Cave) to your hand to pay the cost of Wonderscape Sage's last ability, you won't discard a card.

I believe this now works, I copied the ReturnToHandChosenControlledPermanentCost and just added the nonbasic subtype check in there and then stored the out come to be checked in the effect

**Comment by JayDi85** [comment]
[[Wonderscape Sage]]

File: Mage.Sets/src/mage/cards/w/WonderscapeSage.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                }
+                player.discardOne(false, false, source, game);
+                return true;
+            }
+        }
+        return true;
+    }
+}
+
+class WonderscapeSageCost extends CostImpl {
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/w/WonderscapeSage.java:107]: No need in custom cost. Update original cost instead (see attached screen with such cost) to save permanent, see example from SacrificeTargetCost:
IMG_0545

Then search payed permanent from costs list:
IMG_0546

Comment (grimreap124) [reply Mage.Sets/src/mage/cards/w/WonderscapeSage.java:107]: Awesome I can do that tomorrow

Comment (JayDi85) [reply Mage.Sets/src/mage/cards/w/WonderscapeSage.java:107]: Original problem with Omo, Queen of Vesuva is out of scope of current PR, see #12506 and related discussion. Just ignore that problem here.

File: Mage.Sets/src/mage/cards/w/WonderscapeSage.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // {T}, Return a land you control to its owner's hand: Draw a card. Then discard a card unless that land had a nonbasic land type.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new WonderscapeSageEffect(), new TapSourceCost());
+        ability.addCost(new ReturnToHandChosenControlledPermanentCost(new TargetControlledPermanent(filter)));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/w/WonderscapeSage.java:54]: StaticFilters.FILTER_CONTROLLED_PERMANENT_LAND, and then you don't need either filter

**Pull request closed without merging**